boot/boot_serial/src/boot_serial.c

```diff
@@ -312,10 +313,14 @@ bs_upload(char *buf, int len)
         rc = 0;
         goto out;
     }
-    if (curr_off + img_blen < img_size) {
-        rem_bytes = img_blen % flash_area_align(fap);
+
+    rem_bytes = img_blen % flash_area_align(fap);
+
+    if (((curr_off + img_blen < img_size) && rem_bytes) ||
+        (curr_off + img_blen > img_size)) {
```

Comment:
It this gets true, but `rem_bytes == 0`, then it still should be an error, right?

Reply:
Lets assume:
```
img_blen = 16
curr_off = 4
img_size = 16
flash_area_align(fap) = 4
rem_bytes = 16 % 4 => rem_bytes = 0;
```
now
```
cur_off + img_len > img_size => 4 + 16 > 16 is true
```
but `rem_bytes == 0` which means that
```
if (rem_bytes) {
        img_blen -= rem_bytes;
        rc = MGMT_ERR_EINVAL;
        goto out;
}
```
will not happen because `rem_bytes == 0`

Reply:
Indeed ,I forgoten about the line below - It has been tested on another branch where it was removed.